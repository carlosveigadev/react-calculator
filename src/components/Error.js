import Navbar from './Navbar';

const Error = () => {
  <>
    <Navbar />
    <h1>Oops! Page not found!</h1>
  </>;
};

export default Error;
