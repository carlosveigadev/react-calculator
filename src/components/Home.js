import Navbar from './Navbar';
import styles from '../styles/style.module.css';

const Home = () => (
  <>
    <Navbar />
    <div className={styles.home}>
      <h1>Welcome to our page!</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Aliquid rerum maxime commodi ea, neque modi voluptates cum mollitia
        odio maiores ducimus earum
        ad quo perspiciatis ab soluta ipsum fugit consequuntur consectetur,
        quisquam nesciunt officia magni!
        Esse animi suscipit, doloremque quas nostrum dolore, architecto
        quae nihil aperiam consectetur pariatur
        sapiente sed cumque enim, accusantium ipsam labore natus nam sequi reiciendis.
        Vel cum alias provident rerum doloremque impedit dicta at voluptatibus
        earum fuga commodi unde repudiandae natus placeat vitae perspiciatis laboriosam
        omnis similique facere quos, deleniti officiis! Eligendi labore sunt, illo id
        ipsam minima architecto animi? Id impedit earum iure nesciunt.
        Voluptatem magni, amet similique illum quia unde tenetur
        dolores in labore minus facere nostrum ab veniam quam hic. Sed velit sit totam unde!
        Officiis veritatis quas, velit illum sint voluptatem quasi fuga,
        ea quidem reiciendis eum dolore iusto atque dicta
        harum dignissimos. Ducimus ab magni accusantium commodi, molestias
        architecto consequuntur voluptate dolorum a modi omnis nulla culpa aliquam velit.
        Eveniet quae facilis debitis voluptate iure ad quasi mollitia quos quas,
        libero modi nihil incidunt atque vitae fugiat eum voluptas laudantium eius
        consequuntur earum eaque sint sequi blanditiis expedita.
        Assumenda optio porro dicta
        repudiandae consequatur. Aliquid culpa officia reprehenderit impedit neque qui?

      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Aliquid rerum maxime commodi ea, neque modi voluptates cum mollitia
        odio maiores ducimus earum
        ad quo perspiciatis ab soluta ipsum fugit consequuntur consectetur,
        quisquam nesciunt officia magni!
        Esse animi suscipit, doloremque quas nostrum dolore, architecto
        quae nihil aperiam consectetur pariatur
        sapiente sed cumque enim, accusantium ipsam labore natus nam sequi reiciendis.
        Vel cum alias provident rerum doloremque impedit dicta at voluptatibus
        earum fuga commodi unde repudiandae natus placeat vitae perspiciatis laboriosam
        omnis similique facere quos, deleniti officiis! Eligendi labore sunt, illo id
        ipsam minima architecto animi? Id impedit earum iure nesciunt.
        Voluptatem magni, amet similique illum quia unde tenetur
        dolores in labore minus facere nostrum ab veniam quam hic. Sed velit sit totam unde!
        Officiis veritatis quas, velit illum sint voluptatem quasi fuga,
        ea quidem reiciendis eum dolore iusto atque dicta
        harum dignissimos. Ducimus ab magni accusantium commodi, molestias
        architecto consequuntur voluptate dolorum a modi omnis nulla culpa aliquam velit.
        Eveniet quae facilis debitis voluptate iure ad quasi mollitia quos quas,
        libero modi nihil incidunt atque vitae fugiat eum voluptas laudantium eius
        consequuntur earum eaque sint sequi blanditiis expedita.
        Assumenda optio porro dicta
        repudiandae consequatur. Aliquid culpa officia reprehenderit impedit neque qui?
      </p>
    </div>

  </>
);

export default Home;
