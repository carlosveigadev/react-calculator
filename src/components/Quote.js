const Quote = () => (
  <>
    <h1>
      Mathematics is not about numbers, equations, computations, or algorithms:
      it is about understanding. â€“William Paul Thurston
    </h1>
  </>
);

export default Quote;
