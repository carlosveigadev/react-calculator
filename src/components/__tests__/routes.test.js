import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';
import Calculator from '../App';
import Quote from '../Quote';
import Error from '../Error';

describe('Check if routes work ', () => {
  it('renders Home page correctly in the proper route.', () => {
    const home = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,

      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });

  it('renders Quote page correctly in the proper route.', () => {
    const quote = renderer
      .create(
        <BrowserRouter>
          <Quote />
        </BrowserRouter>,

      )
      .toJSON();
    expect(quote).toMatchSnapshot();
  });

  it('renders App page correctly in the proper route.', () => {
    const app = renderer
      .create(
        <BrowserRouter>
          <Calculator />
        </BrowserRouter>,

      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });

  it('renders error page correctly in the proper route.', () => {
    const error = renderer
      .create(
        <BrowserRouter>
          <Error />
        </BrowserRouter>,

      )
      .toJSON();
    expect(error).toMatchSnapshot();
  });
});
