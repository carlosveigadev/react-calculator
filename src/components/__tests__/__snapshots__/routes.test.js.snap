// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if routes work  renders App page correctly in the proper route. 1`] = `
Array [
  <div
    className="navbar"
  >
    <h2>
      Math Magicians
    </h2>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>,
  <div
    className="flex"
  >
    <div
      className="text"
    >
      <h1>
        Let's do some math!
      </h1>
    </div>
    <div
      className="app"
    >
      <div
        className="display"
      >
        <h1
          data-testid="display"
        >
          0
        </h1>
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className="button"
          data-testid="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          data-testid="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          data-testid="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button buttonOrange"
          data-testid="÷"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className="button"
          data-testid="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          data-testid="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          data-testid="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button buttonOrange"
          data-testid="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className="button"
          data-testid="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          data-testid="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          data-testid="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button buttonOrange"
          data-testid="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className="button"
          data-testid="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          data-testid="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          data-testid="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button buttonOrange"
          data-testid="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className="button button50"
          data-testid="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          data-testid="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button buttonOrange"
          data-testid="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Check if routes work  renders Home page correctly in the proper route. 1`] = `
Array [
  <div
    className="navbar"
  >
    <h2>
      Math Magicians
    </h2>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>,
  <div
    className="home"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid rerum maxime commodi ea, neque modi voluptates cum mollitia odio maiores ducimus earum ad quo perspiciatis ab soluta ipsum fugit consequuntur consectetur, quisquam nesciunt officia magni! Esse animi suscipit, doloremque quas nostrum dolore, architecto quae nihil aperiam consectetur pariatur sapiente sed cumque enim, accusantium ipsam labore natus nam sequi reiciendis. Vel cum alias provident rerum doloremque impedit dicta at voluptatibus earum fuga commodi unde repudiandae natus placeat vitae perspiciatis laboriosam omnis similique facere quos, deleniti officiis! Eligendi labore sunt, illo id ipsam minima architecto animi? Id impedit earum iure nesciunt. Voluptatem magni, amet similique illum quia unde tenetur dolores in labore minus facere nostrum ab veniam quam hic. Sed velit sit totam unde! Officiis veritatis quas, velit illum sint voluptatem quasi fuga, ea quidem reiciendis eum dolore iusto atque dicta harum dignissimos. Ducimus ab magni accusantium commodi, molestias architecto consequuntur voluptate dolorum a modi omnis nulla culpa aliquam velit. Eveniet quae facilis debitis voluptate iure ad quasi mollitia quos quas, libero modi nihil incidunt atque vitae fugiat eum voluptas laudantium eius consequuntur earum eaque sint sequi blanditiis expedita. Assumenda optio porro dicta repudiandae consequatur. Aliquid culpa officia reprehenderit impedit neque qui?
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid rerum maxime commodi ea, neque modi voluptates cum mollitia odio maiores ducimus earum ad quo perspiciatis ab soluta ipsum fugit consequuntur consectetur, quisquam nesciunt officia magni! Esse animi suscipit, doloremque quas nostrum dolore, architecto quae nihil aperiam consectetur pariatur sapiente sed cumque enim, accusantium ipsam labore natus nam sequi reiciendis. Vel cum alias provident rerum doloremque impedit dicta at voluptatibus earum fuga commodi unde repudiandae natus placeat vitae perspiciatis laboriosam omnis similique facere quos, deleniti officiis! Eligendi labore sunt, illo id ipsam minima architecto animi? Id impedit earum iure nesciunt. Voluptatem magni, amet similique illum quia unde tenetur dolores in labore minus facere nostrum ab veniam quam hic. Sed velit sit totam unde! Officiis veritatis quas, velit illum sint voluptatem quasi fuga, ea quidem reiciendis eum dolore iusto atque dicta harum dignissimos. Ducimus ab magni accusantium commodi, molestias architecto consequuntur voluptate dolorum a modi omnis nulla culpa aliquam velit. Eveniet quae facilis debitis voluptate iure ad quasi mollitia quos quas, libero modi nihil incidunt atque vitae fugiat eum voluptas laudantium eius consequuntur earum eaque sint sequi blanditiis expedita. Assumenda optio porro dicta repudiandae consequatur. Aliquid culpa officia reprehenderit impedit neque qui?
    </p>
  </div>,
]
`;

exports[`Check if routes work  renders Quote page correctly in the proper route. 1`] = `
Array [
  <div
    className="navbar"
  >
    <h2>
      Math Magicians
    </h2>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>,
  <h1
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
  </h1>,
]
`;

exports[`Check if routes work  renders error page correctly in the proper route. 1`] = `
Array [
  <div
    className="navbar"
  >
    <h2>
      Math Magicians
    </h2>
    <div
      className="links"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>,
  <h1
    className="error"
  >
    Oops! Page not found!
  </h1>,
]
`;
